nengo/params.py

```diff
@@ -304,9 +304,7 @@ def __init__(self, name, default=Unconfigurable, length=None, low=0,
         self.low = low
 
     def __set__(self, instance, value):
-        try:
-            value = tuple(int(v) for v in value)
-        except TypeError:
+        if not all(is_integer(v) for v in value):
```

Comment:
> You wouldn't have that convenience in the future anyways

I wouldn't necessarily have that convenience in Numpy (though it might be quite a while before they actually remove that), but I would still have it here.

So I think `floor` and `ceil` remove any rounding errors for reasonably sized numbers, but to really make a robust system we'd need to check tolerances or something, and that's all getting too complex. So I'm fine having to be explicit. That way, there's no magic going on, making it easier for users to debug.